app/views/idv/shared/_warning.html.erb

```diff
@@ -0,0 +1,36 @@
+<%#
+yields: Page instruction content.
+locals:
+* heading: Page title and primary page heading text.
+* action: Optional hash of `text`, `url` of a primary action link.
+* options: Optional proc to modify default set of options.
```

Comment:
I'd call this `modify_options`, since it's not providing options directly?

Reply:
Also, looks like we only use it to prepend, so what if we just passed in a literal array like `prepend_options: [a, b]`

Seeing a `Proc.new` in a template is kind of eyebrow raising for me, since that implies there's moderately complex logic happening, even if in our case it's mostly if-statements

Reply:
> Also, looks like we only use it to prepend, so what if we just passed in a literal array like `prepend_options: [a, b]`

Hm, yeah, I forget the reason I'd thought I'd needed to be able to both _append_ and _prepend_ that motivated this implementation. That being said, my hope in the next pull request is to merge [`_throttled.html.erb`](https://github.com/18F/identity-idp/blob/main/app/views/idv/session_errors/_throttled.html.erb) and this partial into a single partial, where sometimes the "Get help at SP" default option would be the first item ([example](https://www.figma.com/file/PTB99GHGAVtD5YeW69vg05/IAL2-Troubleshooting-options-LG-4309?node-id=252%3A2623)).

A couple thoughts:

- Maybe avoid the "default" options and always expect them to be passed in
- Maybe `prepend_options` and `append_options` as separate options could be fine

I'll also admit I don't fully understand or appreciate the concern with procs, or why they're more complex 🤷

Reply:
>Maybe avoid the "default" options and always expect them to be passed in

Went with this in 7606884.